src/fare-contracts/PurchaseReservation.tsx

```diff
@@ -40,6 +41,13 @@ export const PurchaseReservation: React.FC<Props> = ({reservation}) => {
     }
   };
 
+  // filter out reservations for subaccount
+  const subaccounts = customerProfile?.subAccounts;
+  if (subaccounts) {
+    const subaccountReservation = subaccounts.filter((subaccountId) => subaccountId === reservation.customerAccountId);
+    if (subaccountReservation.length > 0) return null;
+  }
+
```

Comment:
Just a tip that `some` can be used in those situations where first `filter` is used and then checked if length is greater than 0.

```ts
if (customerProfile?.subAccounts?.some((id) => id === reservation.customerAccountId) {
  return null;
}
```

Reply:
Thanks, I will keep this in mind!